kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala

```diff
@@ -198,10 +202,85 @@ class KyuubiSessionManager private (name: String) extends SessionManager(name) {
       ms.registerGauge(EXEC_POOL_ALIVE, getExecPoolSize, 0)
       ms.registerGauge(EXEC_POOL_ACTIVE, getActiveCount, 0)
     }
-    // TODO: support to recover batch sessions with session state store
+    recoverBatchSessions()
     super.start()
   }
 
+  @VisibleForTesting
+  def recoverBatchSessions(): Unit = {
+    val restHost = conf.get(FRONTEND_REST_BIND_HOST)
+      .getOrElse(Utils.findLocalInetAddress.getHostAddress)
+    val restPort = conf.get(FRONTEND_REST_BIND_PORT)
+    val kyuubiInstance = s"$restHost:$restPort"
```

Comment:
`KyuubiRestFrontendService.getConnectionUrl`

Reply:
the frontend service is not started yet.